### ISSUE TITLE
Helm unittest introduction

### ISSUE THREAD

**Issue by chrira** [bitnami/charts#16919]
### Name and Version

all

### What is the problem this feature will solve?

Helm Charts are getting fast complex and contain a lot of functions and logic.

This makes it difficult to work on existing charts. Mostly the features are not documented or not known.
Contributors could break some functionality without knowing it.

I my opinion, developing Helm Charts is like developing source code. This because they contain complex logic. Therefore it should also be tested in the same ways.

There has been a try to start with unit tests before: [[all] Unit tests](https://github.com/bitnami/charts/pull/6986)

### What is the feature you are proposing to solve the problem?

The usage of the [Helm unittest plugin](https://github.com/helm-unittest/helm-unittest) can help deliver better quality and reduce errors.

With small tests existing or new logic could be covered by tests. This will make it easier for others to extend the charts.

Unittests are also a source of feature documentation. There we can see how it should work.

I suggest to introduce helm unittest in every chart. After that any change on the chart template / manifest/ helpers / logic should be covered with tests.
This way we can have unittests without a big initial effort.

Unittest attempts inside bitnami charts:

* [[all] Unit tests](https://github.com/bitnami/charts/pull/6986)
* [[bitnami/haproxy] allow to create ingress](https://github.com/bitnami/charts/pull/12424)
* [[bitnami/sonarqube] enable custom ingress labels](https://github.com/bitnami/charts/pull/16918)

### What alternatives have you considered?

I checked the [Bitnami's Helm Chart Release Process](https://docs.bitnami.com/kubernetes/faq/get-started/understand-charts-release-process/).
This is good for testing the deployed application.
These integration tests do not cover the need for unittests on the chart logic. The feedback loop is also longer.

**Comment by javsalgar** [comment]
Hi!

Thank you so much for the suggestion. Right now, we perform all package, verification and publish using [VMware ImageBuilder](https://tanzu.vmware.com/image-builder). In the [.vib](https://github.com/bitnami/charts/tree/main/.vib) folder you can see all the tests we are performing.

**Comment by chrira** [comment]
Thank you @javsalgar . I looked more in dept in to the actual testing.
They are referenced as deployment tests. I would also say that the actual tests are integration tests.

For extending and refactoring of the Helm Chart itself the ability to write small unit tests would help lots and improve the quality of the charts.

From the Helm documentation:
"Helm has over 60 available functions. Some of them are defined by the [Go template language](https://godoc.org/text/template) itself. Most of the others are part of the [Sprig template library](https://masterminds.github.io/sprig/)."
All the functions and pipelines in the charts are logic that has to be tested. We have good results in our company using helm unittests.

First we just added the test folder and every developer could execute tests locally. The idea was to add tests for new logic. Then we automated the test execution with a build server.

What would be nice is the okey from the core contributors to start with helm unittesting.
